storage/tsdb/tsdb.go

```diff
@@ -139,6 +147,50 @@ func Open(path string, l log.Logger, r prometheus.Registerer, opts *Options) (*t
 	return db, nil
 }
 
+// StartTime implements the Storage interface.
+func (a adapter) StartTime() (int64, error) {
+	startTime := int64(model.Latest)
+
+	var indexr tsdb.IndexReader
+	if len(a.db.Blocks()) > 0 {
+		indexr = a.db.Blocks()[0].Index()
+	} else {
+		indexr = a.db.Head().Index()
+	}
+
+	joblabel := "job"
+	tpls, err := indexr.LabelValues(joblabel)
+	if err != nil {
+		return startTime, err
+	}
+
+	for i := 0; i < tpls.Len(); i++ {
+		vals, err := tpls.At(i)
+		if err != nil {
+			continue
+		}
+
+		for _, v := range vals {
+			p, err := indexr.Postings(joblabel, v)
+			if err != nil {
+				continue
+			}
+
+			if p.Next() {
+				var lset tsdbLabels.Labels
+				var chks []tsdb.ChunkMeta
+				indexr.Series(p.At(), &lset, &chks)
+				if startTime > chks[0].MinTime {
+					startTime = chks[0].MinTime
+				}
+			}
+		}
+	}
+	// Add a safety margin as it may take a few minutes for everything to spin up.
```

Comment:
Data older than 5m could be ingested, causing problems here. The --storage.tsdb.min-block-duration flag is probably what we need here.